docs/source/federated-types/sharing-types.mdx

```diff
@@ -270,6 +335,122 @@ And here's the problem: if Subgraph B _doesn't_ define `Query.currentPosition`,
 
 **Composition recognizes this potential problem, and it fails with an error.** So how do we fix it? Check out [Solutions for unresolvable fields](./composition/#solutions-for-unresolvable-fields).
 
+## Adding new shared fields
+
+Adding a new field to a value type can cause composition issues, because it's challenging to add the field to _all_ defining subgraphs at the same time.
+
+Let's say we're adding a `z` field to our `Position` value type, and we start with Subgraph A:
+
+<p style="margin-bottom: 0">⚠️</p>
+
+<CodeColumns>
+
+```graphql {4} title="Subgraph A"
+type Position @shareable {
+  x: Int!
+  y: Int!
+  z: Int!
+}
+```
+
+```graphql title="Subgraph B"
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+
+</CodeColumns>
+
+It's likely that when we attempt to compose these two schemas, composition will _fail_, because Subgraph B can't resolve `Position.z`.
+
+To incrementally add the field to _all_ of our subgraphs _without_ breaking composition, we can use the [`@inaccessible` directive](#using-inaccessible).
+
+
+### Using `@inaccessible`
+
+If you apply the `@inaccessible` directive to a field, composition _omits_ that field from your supergraph schema. This helps you incrementally add a field to _multiple_ subgraphs without breaking composition.
+
+To use `@inaccessible` in a subgraph, first make sure you include it in the `import` array of your Federation 2 opt-in declaration:
+
+```graphql {3} title="Subgraph A"
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: ["@key", "@shareable", "@inaccessible"])
+```
+
+Then, whenever you add a new field to a value type, apply `@inaccessible` to that field if it isn't yet present in _every_ subgraph that defines the value type:
+
+<CodeColumns>
+
+```graphql {4} title="Subgraph A"
+type Position @shareable {
+  x: Int!
+  y: Int!
+  z: Int! @inaccessible
+}
+```
+
+```graphql title="Subgraph B"
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+
+</CodeColumns>
+
+> Even if `Position.z` is defined in _multiple_ subgraphs, you only need to apply `@inaccessible` in _one_ subgraph to omit it. In fact, you usually _should_ only apply it in one subgraph to simplify removing it later.
+
+With this syntax, composition ignores `Position.z`, and the resulting `Position` type includes only `x` and `y` fields.
```

Comment:
Nit: technically, composition doesn't "ignore" `Position.z`: if you have 2 definitions of an inaccessible field that have incompatible, composition will still actually error out. It's that because `z` is not exposed in the supergraph API, it cannot be queried and so composition won't complain that "if it was queried, it couldn't be reached in the other subgraphs".